pyfar/dsp/dsp.py

```diff
@@ -1488,3 +1488,232 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
             " or 'freq_raw'.")
     data[data == 0] = np.finfo(float).eps
     return log_prefix * np.log10(np.abs(data) / log_reference)
+
+
+def soft_limit_spectrum(signal, limit, knee, ratio=np.inf, freq_range=None,
+                        direction='above'):
+    """
+    Soft limit spectrum of a Signal or FrequencyData object.
+
+    Apply a frequency-wise soft limiting to a spectrum. Soft limiting denotes
+    a smooth transition that is applied to limit discontinuities in the data
+    that would appear in hard clipping. The transition phase is termed `knee`,
+    and two different types of knees can be applied (see parameters and
+    examples below).
+
+    Note that the limiting is applied on `signal.freq`, i.e. the data after
+    the FFT normalization in case `signal` is a `pyfar.Signal` object.
+
+    Parameters
+    ----------
+    signal : Signal, FrequencyData
+        The input data
+    limit : number, array like
+        The gain in dB at which the limiting reaches its full eeffect. If this
+        is a number, the same limit is applied to all frequencies. If this an
+        an array like, it must be broadcastable to `signal.freq.shape`.
+    knee : number, string
+        ``'arctan'``
+            Apply an arcus tanges knee according to [#]_ Section 3.6.4. In this
+            case the parameter `ratio` (below) is ignored. This knee definition
+            originates from signal mircorphone array signal processing.
+        number
```

Comment:
Maybe "width" or "value" is a better name for this option?

Reply:
hm, this means that you can either pass 'arctan' or a number. Is there a good way to make it clearer?

Reply:
What do you think of, instead having the two options in a list, combine these into sentences. That it might be clearer, that "number" refers to a value and is not an option?

Reply:
Changing the order between the two could also help in this case.